Clear filters

one1color opened the pull request:
fixes #25

nmondollot commented:
Great, it works 👍 

My only nitpick: when I slide to the right to see the favorite/project filters, it's a bit weird when I tap on them.
The buttons move to the right and I don't see the clear button

See video:


https://github.com/user-attachments/assets/84ec3d89-b3a0-4e47-8ba5-af75370b87e0

On Android @wang-li worked on a more elegant solution.
Can you test it and see if we could do something similar?

nmondollot commented:
> When user tap the the popular button, it will cancel, selected circuit and applied filter is canceled. I understand to cancel circuit selection, but probably people wants to see the popular problems within selected levels?

They can if they choose the popular filter first, and then choose the levels.
But I agree it's a bit confusing.
I think I did it this way to make it easy to switch between the 2 modes (popular and levels) without having to clear the filters.
But if we have a nice clear button we can definitely make it simpler (I think it's this way on Android)

wang-li commented:
> On Android @wang-li worked on a more elegant solution.
Can you test it and see if we could do something similar?

@one1color Here is a screen recording of the clear filters button interaction that is currently implemented on Android. It has the benefit of being accessible at any time and any scroll position of the filters.

https://github.com/user-attachments/assets/6d9094fe-b3b0-4a23-a89c-060419deacb5

one1color commented:
I have updated codes so that the clear button is fixed when it appears. Currently no animation or something applied. let me know if it's better having it. 

https://github.com/user-attachments/assets/51d7f09f-fed3-49a0-8ab4-009f97f990f6

one1color commented:
> I have updated codes so that the clear button is fixed when it appears. Currently no animation or something applied. let me know if it's better having it. @nmondollot 
> 
>  Simulator.Screen.Recording.-.iPhone.16.Pro.-.2024-11-21.at.16.21.03.1.mp4

nmondollot commented:
Hello @one1color 
Sorry it took me so long to answer! 😓 

I tested your branch, thank you 🙏 

2 things:

1) I find these margins a bit ugly:

<img width="440" alt="Capture d’écran 2025-01-08 à 16 24 14" src="https://github.com/user-attachments/assets/eacf0927-1385-4281-9033-2bdaf0e5ab90" />

Is there a way to make the scrollview full width when the clear button is not present?

2) When the clear button is present we could use the same kind of vertical delimiter as on Android, to avoid sharp edged (again, I find them ugly)

<img width="165" alt="Capture d’écran 2025-01-08 à 16 28 37" src="https://github.com/user-attachments/assets/5e058f95-3271-4325-8456-12d29c47608f" />

<img width="148" alt="Capture d’écran 2025-01-08 à 16 28 10" src="https://github.com/user-attachments/assets/769acf6b-a480-47cb-bcc4-4596e6c14a19" />


FYI I'm going on a 1.5-month vacation, so I won't be available until the end of february!
I will publish a new version when I get back (with the new mapbox v11 code)

one1color commented:
Hey! @nmondollot  It's been so long since I replied the message last time. So sorry! As you can see the video, I have removed the margins of right side, and also, I have added some animation to appear/disappear the x button smoothly. 

About the vertical delimiter, the way i came up was using `geometoryReader` which i would like to avoid to use, and will be bit complex as it should be for now imo.
If you are fine with this, I would merge this PR, and adding it as issues to further improvement.

https://github.com/user-attachments/assets/b3fbc062-d034-4167-b985-8b1240950b0b
